metpy/calc/thermo.py

```diff
@@ -1711,3 +1711,155 @@ def thickness_hydrostatic_from_relative_humidity(pressure, temperature, relative
 
     return thickness_hydrostatic(pressure, temperature, mixing=mixing, bottom=bottom,
                                  depth=depth)
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def brunt_vaisala_frequency_squared(heights, potential_temperature, axis=0):
+    r"""Calculate the square of the Brunt-Vaisala frequency.
+
+    Brunt-Vaisala frequency squared (a measure of atmospheric stability) is given by the
+    formula:
+
+    .. math:: N^2 = \frac{g}{\theta} \frac{d\theta}{dz}
+
+    This formula is based off of Equations 3.75 and 3.77 in [Hobbs2006]_.
+
+    Parameters
+    ----------
+    heights : array-like
+        One-dimensional profile of atmospheric height
+    potential_temperature : array-like
+        Atmospheric potential temperature
+    axis : int, optional
+        The axis corresponding to vertical in the potential temperature array, defaults to 0.
+
+    Returns
+    -------
+    array-like
+        The square of the Brunt-Vaisala frequency.
+
+    See Also
+    --------
+    brunt_vaisala_frequency, brunt_vaisala_frequency_layer, brunt_vaisala_period,
+    potential_temperature
+
+    """
+    # Ensure validity of temperature units
+    potential_temperature.ito('K')
+
+    # Calculate and return the square of Brunt-Vaisala frequency
+    return g / potential_temperature * first_derivative(potential_temperature, x=heights,
+                                                        axis=axis)
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def brunt_vaisala_frequency(heights, potential_temperature, axis=0):
+    r"""Calculate the Brunt-Vaisala frequency.
+
+    This function will calculate the Brunt-Vaisala frequency as follows:
+
+    .. math:: N = \left( \frac{g}{\theta} \frac{d\theta}{dz} \right)^\frac{1}{2}
+
+    This formula based off of Equations 3.75 and 3.77 in [Hobbs2006]_.
+
+    This function is a wrapper for `brunt_vaisala_frequency_squared` that filters out negative
+    (unstable) quanties and takes the square root.
+
+    Parameters
+    ----------
+    heights : array-like
+        One-dimensional profile of atmospheric height
+    potential_temperature : array-like
+        Atmospheric potential temperature
+    axis : int, optional
+        The axis corresponding to vertical in the potential temperature array, defaults to 0.
+
+    Returns
+    -------
+    array-like
+        Brunt-Vaisala frequency.
+
+    See Also
+    --------
+    brunt_vaisala_frequency_layer, brunt_vaisala_frequency_squared, brunt_vaisala_period,
+    potential_temperature
+
+    """
+    bv_freq_squared = brunt_vaisala_frequency_squared(heights, potential_temperature,
+                                                      axis=axis)
+    bv_freq_squared[bv_freq_squared.magnitude < 0] = np.nan
+
+    return np.sqrt(bv_freq_squared)
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def brunt_vaisala_frequency_layer(heights, potential_temperature, axis=0):
```

Comment:
I wonder about this one. I don't know if a simple mean is appropriate--or if it is, if the function is worth incorporating. I see the original GEMPAK mentions `LAV`. @kgoebber can you shed light on what GEMPAK's `LAV` does? Can either of you describe the use case for including a layer-average BV?

We already have `mixed_layer` for doing mass-weighted averages, and I'm wondering if we should just punt people to manually do that rather than include layer-average version of all the functions. I'm worried about the combinatorial explosion of trying to include functions for all variants vs. properly compassable parts.

Reply:
Thank you for catching that, on closer thought, I don't think a simple mean is appropriate either.

Layer-average BV is how GEMPAK does it, which is why this `_layer` function was implemented (based on the discussion in #629). But, each of the functions in this bunch appear to be layer quantities in GEMPAK as well...so frequency squared and period would also have this problem. I don't recall a specific use case for the layer-average BV from my course experience, only that the layer version is the only way to get it from GEMPAK.

(Also, in case it is useful, I was also curious about how `LAV` worked, and here is what I found in the docs: `LAV (S) = [ S (level1) + S (level2) ] / 2.`)

Would the best way forward then be to remove `brunt_vaisala_frequency_layer`, and just mention in the GEMPAK table to use these functions in combination with `mixed_layer` for a layer average? Or mention somehow that our implementation approach differs from GEMPAK's in calculating the frequency (or period) on each level rather than a layer between two levels?

Reply:
Wow, `LAV` is...stupidly simple. I lean towards the removal of the `_layer` version and just point to `mixed_layer` for the average. And if someone *really* wants the `LAV` behavior, we can point them to `numpy.mean`.  Thoughts @kgoebber @jrleeman ?